ext/storage_sources/s3_store/s3_connection.cpp

```diff
@@ -43,7 +43,7 @@ S3Connection::ListObjects(const std::string &prefix, std::vector<std::string> &o
         return (1);
     auto result = outcomes.GetResult();
     for (const auto &object : result.GetContents())
-        objects.push_back(object.GetKey());
+        objects.push_back(object.GetKey().substr(_objectPrefix.length()));
```

Comment:
Please add a comment explaining that we are stripping the listing of the prefix configured for the file system.